### PR TITLE
chore: Refactor the implementation of getConnectedDevices helper

### PR DESCRIPTION
**Pull request by mykola-mokhnach**

### CODE REVIEW

File: lib/tools/system-calls.js [hunk 1]
```diff
@@ -153,34 +153,38 @@ systemCallMethods.getBinaryFromPath = async function getBinaryFromPath (binaryNa
...
   }
+  const listHeader = 'List of devices';
+  // expecting adb devices to return output as
+  // List of devices attached
+  // emulator-5554	device
+  const startingIndex = stdout.indexOf(listHeader);
+  if (startingIndex < 0) {
+    throw new Error(`Unexpected output while trying to get devices. Output was: ${stdout}`);
+  }
+  // slicing ouput we care about
```
Comment (imurchie) [inline lib/tools/system-calls.js]: output

File: lib/tools/system-calls.js [hunk 2]
```diff
@@ -153,34 +153,38 @@ systemCallMethods.getBinaryFromPath = async function getBinaryFromPath (binaryNa
...
+  stdout = stdout.slice(startingIndex);
+  const excludedLines = [listHeader, 'adb server', '* daemon', 'offline'];
+  const devices = stdout.split('\n')
+    .map((line) => line.trim())
```
Comment (imurchie) [inline lib/tools/system-calls.js]: this line can also just be `.map(_.trim)`

File: lib/tools/system-calls.js (hunk 1 above)
Comment (mykola-mokhnach) [reply lib/tools/system-calls.js]: fixed

File: lib/tools/system-calls.js (hunk 2 above)
Comment (mykola-mokhnach) [reply lib/tools/system-calls.js]: changed

**Pull request merged**